python/pyspark/sql/tests/test_arrow_python_udf.py

```diff
@@ -219,6 +220,17 @@ def test_udf(a, b):
         with self.assertRaises(PythonException):
             self.spark.sql("SELECT test_udf(id, a => id * 10) FROM range(2)").show()
 
+    def test_udt_as_return_type(self):
+        data = [
+            ExamplePoint(1.0, 2.0),
+        ]
+        schema = StructType().add("point", ExamplePointUDT())
+        df = self.spark.createDataFrame([data], schema=schema)
+        [row] = df.select(
+            udf(lambda x: x, returnType=ExamplePointUDT(), useArrow=True)("point"),
```

Comment:
So the data inside the UDF execution is ArrayType of Doubles instead of `ExamplePoint`, is this correct?

Reply:
> So the data inside the UDF execution is ArrayType of Doubles instead of `ExamplePoint`, is this correct?

For current code, data inside the UDF execution is ArrayType of Doubles, this PR for output checking,  still checking how to resolve this.
But seems toPandas API also store ArrayType of Doubles, not ExamplePoint?
Also `toArrowTypes`

Reply:
What about UDF without Arrow optimization?

Reply:
> What about UDF without Arrow optimization?

```
        df.select(
            udf(lambda x: type(x), returnType=StringType(), useArrow=True)("point"),
        ).show(1, False)
        df.select(
            udf(lambda x: type(x), returnType=StringType(), useArrow=False)("point"),
        ).show(1, False)
```
Result is 
```
+---------------+
|<lambda>(point)|
+---------------+
|<class 'list'> |
+---------------+

+----------------------------------------------------------+
|<lambda>(point)                                           |
+----------------------------------------------------------+
|net.razorvine.pickle.objects.ClassDictConstructor@12fc91b3|
+----------------------------------------------------------+

```

Reply:
> What about UDF without Arrow optimization?

Maybe we should implement an UDTArrowWriter?

Reply:
I think we should at least construct the UDT type instance to match with non-arrow-optimized one.

Reply:
The root cause is `pa.Table.from_batches([batch]).itercolumns()` cann't pass UDT, so we won't call UDT's deserialize when `load_stream`

Reply:
`toPandas()` is correct because it directly know the field type when call `toPandas()`

Reply:
What happens if the UDF explicitly returns `ExamplePoint`?

Reply:
> What happens if the UDF explicitly returns `ExamplePoint`?

```
       def eval(x) -> ExamplePoint:
            return x

        df.select(
            udf(eval, returnType=BooleanType(), useArrow=False)("point"),
        ).show(1, False)

        df.select(
            udf(eval, returnType=BooleanType(), useArrow=True)("point"),
        ).show(1, False)



Without arrow return 
+-----------+
|eval(point)|
+-----------+
|null       |
+-----------+

With arrow throw
pyarrow.lib.ArrowInvalid: Could not convert [1.0, 2.0] with type list: tried to convert to boolean

```